api/src/main/java/com/epam/pipeline/manager/execution/PipelineLauncher.java

```diff
@@ -103,40 +105,42 @@ public class PipelineLauncher {
     private final SimpleDateFormat dateFormat = new SimpleDateFormat(Constants.SIMPLE_DATE_FORMAT);
     private final SimpleDateFormat timeFormat = new SimpleDateFormat(Constants.SIMPLE_TIME_FORMAT);
 
-    public String launch(PipelineRun run, PipelineConfiguration configuration, List<String> endpoints,
-                         String nodeIdLabel, String clusterId) {
-        return launch(run, configuration, endpoints, nodeIdLabel, true, run.getPodId(), clusterId);
+    @PipelineLaunchCheck
+    public String launch(final PipelineRun run, final PipelineConfiguration configuration,
+                         final List<String> endpoints, final String clusterId) {
+        return launch(run, configuration, endpoints, true, run.getPodId(), clusterId);
     }
 
-    public String launch(PipelineRun run, PipelineConfiguration configuration,
-                         List<String> endpoints, String nodeIdLabel, boolean useLaunch,
-                         String pipelineId, String clusterId) {
-        return launch(run, configuration, endpoints, nodeIdLabel, useLaunch, pipelineId, clusterId,
-                getImagePullPolicy(configuration));
+    @PipelineLaunchCheck
+    public String launch(final PipelineRun run, final PipelineConfiguration configuration, final List<String> endpoints,
+                         final boolean useLaunch, final String pipelineId, final String clusterId) {
+        return launch(run, configuration, endpoints, useLaunch, pipelineId,
+                clusterId, getImagePullPolicy(configuration));
     }
 
-    public String launch(PipelineRun run, PipelineConfiguration configuration,
-                         List<String> endpoints, String nodeIdLabel, boolean useLaunch,
-                         String pipelineId, String clusterId, ImagePullPolicy imagePullPolicy) {
-        GitCredentials gitCredentials = configuration.getGitCredentials();
+    @PipelineLaunchCheck
```

Comment:
I'm not sure if we really need an annotation and aspect for validation, as in fact we have one method with full list of arguments and we can just add the check to its code?

Reply:
Indeed, you are right.